src/cdomains/intDomain.ml

```diff
@@ -1826,9 +1925,467 @@ module Enums : S with type int_t = BigInt.t = struct
       10, QCheck.map pos (ISet.arbitrary ());
     ] (* S TODO: decide frequencies *)
 
+    let refine_with_congruence ik a b =
+      let contains c m x = if BI.equal m BI.zero then BI.equal c x else (BI.rem (BI.sub x c) m) == BI.zero in
+      match a with
+      | Inc e -> ( match b with
+                | None -> Inc e
+                | Some (c, m) -> Inc (ISet.filter (contains c m) e))
+      | a -> a
+  let refine_with_interval ik a b = a
+  let refine_with_excl_list ik a b = match a, b with
+  | _, None -> a
+  | a , Some (ls) -> meet ik a (of_excl_list ik ls)
+  let refine_with_incl_list ik a b = match a, b with
+  | _, None -> a
+  | Inc x, Some (ls) -> meet ik (Inc x) (Inc (ISet.of_list ls))
+  | a, _ -> a
 
 end
 
+module CongruenceFunctor(Ints_t : IntOps.IntOps): S with type int_t = Ints_t.t and type t = (Ints_t.t * Ints_t.t) option =
```

Comment:
I guess this doesn't need to be parametric in `Ints_t`, we could just use `BigIntOps` here.

Reply:
This is again so because we initially copied the code from the interval domain